### PR TITLE
Increase precision of dot bounds check in EGP:DrawPath

### PR DESCRIPTION
**Pull request by Denneisk**
Fixes a bug with EGP CircleOutlines where the dot product could be >0.999 and <=1, causing the function to skip calculating every segment but the last.

As far as I can tell, having these so imprecise isn't actually necessary.

### CODE REVIEW

**Review by thegrb93 — CHANGES_REQUESTED** [review]
Can you change the comparators to >= and <=? Also move the local scaling calculation?

```lua
if dot >= 1 then
    ...
elseif dot <= -1 then
    ...
else
    local scaling = size*math.tan(math.acos(dot)/2)
    if dir.x*-lastdir.y + dir.y*lastdir.x > 0 then
        ...
    else
        ...
    end
end
```

File: lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua [hunk 1]
```diff
@@ -348,34 +348,37 @@ function EGP:DrawPath( vertices, size, closed )
...
+									if dot < 0 then
+										corners[#corners+1] = { l={x=x1+offsetx, y=y1+offsety}, r={x=x1+(lastdir.x-lastdir.y)*size, y=y1+(lastdir.y+lastdir.x)*size} }
+										corners[#corners+1] = { l={x=x1+offsetx, y=y1+offsety}, r={x=x1-(dir.x+dir.y)*size, y=y1-(dir.y-dir.x)*size} }
+									else
+										corners[#corners+1] = { l={x=x1+offsetx, y=y1+offsety}, r={x=x1-offsetx, y=y1-offsety} }
+									end
 								end
 							end
 						end
+					end
```
Comment (thegrb93) [inline lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua:381]: One too many 'end'?

**Comment by Denneisk** [comment]
The function works fine at +/-1 but I'll oblige in case there's some weird precision issue.

**Review by thegrb93 — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
> The function works fine at +/-1 but I'll oblige in case there's some weird precision issue.

It's supposed to take the simpler branch though as an optimization.

File: lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua (hunk 1 above)
Comment (Denneisk) [reply lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua]: Yeah, must have gotten confused.

**Review by thegrb93 — APPROVED** [review]

**Pull request merged**